src/api/app/controllers/source_package_controller.rb

```diff
@@ -1,6 +1,7 @@
 class SourcePackageController < SourceController
   before_action :require_valid_project_name
-  before_action :require_package, only: %i[show delete]
+  before_action :set_target_project_name, only: %i[show delete]
+  before_action :set_target_package_name, only: %i[show delete]
```

Comment:
These changes introduce a strange behaviour in some other endpoints not being the commands `branch`, `fork` or `release`.

Let's take the `GET /source/:project/:package` endpoint (action `show` of `SourcePackageController`). This endpoint should not take in account the value of a `cmd` query parameter.

If I'm not wrong, with the changes proposed the call `GET /source/some_project/some_package?cmd=branch&target_package=hey` would make the controller assign the string `hey` to `@target_package_name`, wrongly...

Reply:
This controller is not dealing with `cmd` at all...

Reply:
Well, now with the proposed changes, yes it does.

Reply:
Argh, true. :facepalm: :worried: This is wrong.